### PR TITLE
[Perf] flashinfer patch to fuse residual add with following rmsnorm

### PR DESCRIPTION
**Pull request by shashank14k**
## Summary 
The pytorch profile for the flashinfer pipepline showed separate aten::add and flashinfer.norm.rmsnorm launches post attention and post mlp. This PR aims to fuse the two operations into one kernel, reducing the 4 kernels per layer to 2.

### Before
add kernel          ← residual + attn_out
rmsnorm kernel      ← post_attention_layernorm
add kernel          ← residual + mlp_out
rmsnorm kernel      ← next layer's input_layernorm

### After
fused_add_rmsnorm   ← residual + attn_out + post_attention_layernorm
fused_add_rmsnorm   ← residual + mlp_out + next layer's input_layernorm

This required patching the Qwen3Model.forward, because the fused kernel needs both the hidden states and the residual tensor in scope.

### Impact
Number of cuda kernel launches reduced from 9491 to 7699 (10% reduction) and wall time reduced by 6.5% for tested audios. Below are the old and the new pytorch profiles.
[flashinfer_profile_old.txt](https://github.com/user-attachments/files/31846596/flashinfer_profile_old.txt)
[flashinfer_profile_new.txt](https://github.com/user-attachments/files/31846597/flashinfer_profile_new.txt)

## Summary by CodeRabbit

* **Performance**
  * Fused residual and RMS normalization is now enabled by default for optimized model execution.
  * Enabling fused residual normalization automatically enables RMS normalization fusion when needed.
  * Existing attention optimization options remain available.
* **Bug Fixes**
  * Improved layer-specific attention mask selection during optimized execution.
  * Cache behavior now follows the model configuration when not explicitly specified.
  * Added clearer logging for optimization patching activity.

### CODE REVIEW

File: omnivoice/models/omnivoice_flashinfer.py [hunk 1]
```diff
@@ -89,6 +92,147 @@ def _patch_rmsnorm(llm):
...
+        raise ValueError(
+            f"_fi_fused_model_forward expects batch=1 (packed) input, got batch={bsz}"
+        )
+    hidden2d = hidden_states.view(seq, hidden)
+    residual = hidden2d.clone()
+
+    if isinstance(causal_mask_mapping, dict):
+        attn_mask = causal_mask_mapping["full_attention"]
+    else:
+        attn_mask = causal_mask_mapping
```
[the remark this answers is not in the document]
Comment (shashank14k) [reply omnivoice/models/omnivoice_flashinfer.py]: Qwen3-0.6B (OmniVoice) has has_sliding_layers = False — all 28 layers are "full_attention". And the flashinfer packed path bypasses HF's mask entirely via attention_mask={"full_attention": None}, so attn_mask ends up as None regardless.
